### PR TITLE
enhancement(remap): Add to_unix_timestamp function

### LINKED ISSUE #5766: Add VRL function for Unix timestamps
The discussion in PR #5753 produced the idea of adding a function for converting timestamps into Unix timestamps with user-selected precision. Some example uses:

```js
$time = now()

.seconds = to_unix_timestamp($time, unit = "seconds") // The default
.milliseconds = to_unix_timestamp($time, unit = "milliseconds")
.nanoseconds = to_unix_timestamp($time, unit = "nanoseconds")
```

### PR DESCRIPTION
**Pull request by lucperkins**
In response to the discussion in #5730.

Closes #5766

### CODE REVIEW

File: lib/remap-functions/src/to_unix_timestamp.rs [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+        match self {
+            Seconds => "seconds",
+            Milliseconds => "milliseconds",
+            Nanoseconds => "nanoseconds",
+        }
+    }
+}
+
+impl Default for Unit {
```
Comment (jszwedko) [inline lib/remap-functions/src/to_unix_timestamp.rs:67]: Nit: just noting we use the `derivative` crate for defaulting enums in some places: https://mcarton.github.io/rust-derivative/latest/index.html

File: lib/remap-functions/src/to_unix_timestamp.rs [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+            },
+            def: TypeDef {
+                kind: Kind::Integer,
+                ..Default::default()
+            },
+        }
+
+        string_fallible {
+            expr: |_| ToUnixTimestampFn {
+                value: lit!("late December back in '63").boxed(),
```
Comment (jszwedko) [inline lib/remap-functions/src/to_unix_timestamp.rs:143]: Oh, what a night :smile:

**Review by jszwedko — APPROVED** [review]
Nice, thanks @lucperkins !

File: lib/remap-functions/src/to_unix_timestamp.rs (hunk 1 above)
Comment (lucperkins) [reply lib/remap-functions/src/to_unix_timestamp.rs:67]: Good to know! I'm going to forgo that here, as there's another function that does this "the hard way" and that would be out of scope. But I may follow up.

**Pull request merged**